### PR TITLE
Docs review for v2.https

### PR DESCRIPTION
**Pull request by joehan**
### Description
Reviewed the staged HTTPS docs, and found +fixed the following issues:
Formatting Issues: 
- FunctionsErrorCode includes bullet points that were being rendered poorly. Added newlines to fix the formatting
Missing descriptions:
onCall
onRequest
httpError.toJson - Added a brief description
https.HttpsOptions.cors - Added an explanation of what this option does with various types - please double check this to make sure I'm correct
request.rawBody - added a brief description

### CODE REVIEW

File: src/common/providers/https.ts [hunk 1]
```diff
@@ -176,36 +177,52 @@ export interface CallableRequest<T = any> {
  * https://github.com/grpc/grpc/blob/master/doc/statuscodes.md
  *
  * Possible values:
+ *
  * - 'cancelled': The operation was cancelled (typically by the caller).
```
Comment (inlined) [inline src/common/providers/https.ts:181]: Should "cancelled" and others be in code font?

File: src/v2/providers/https.ts [hunk 2]
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
  * Options that can be set on an individual HTTPS Cloud Function.
  */
 export interface HttpsOptions extends Omit<GlobalOptions, 'region'> {
-  /* HTTP functions can override and specify more than one regions. */
+  /** HTTP functions can override global options and can specify multiple regions to deploy to. */
   region?: SupportedRegion | string | Array<SupportedRegion | string>;
+  /** If true, CORS will be allowed for requests to the function.
```
Comment (inlined) [inline src/v2/providers/https.ts:46]: Let's add a comment that `cors` defaults to true for CallableFunctions.

File: src/v2/providers/https.ts [hunk 3]
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
...
 }
 
+/**
+ * Handles HTTPS requests
+ */
 export type HttpsFunction = ((
   req: Request,
   res: express.Response
 ) => void | Promise<void>) & {
   __trigger?: unknown;
```
Comment (inlined) [inline src/v2/providers/https.ts:60]: @alpha to remove docs for this and __endpoint

File: src/v2/providers/https.ts [hunk 4]
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
...
   __endpoint: ManifestEndpoint;
 };
+
+/**
+ * A callable method for clients to call using a Firebase SDK.
+ */
 export interface CallableFunction<T, Return> extends HttpsFunction {
+  /**@hidden */
```
Comment (inlined) [inline src/v2/providers/https.ts:68]: We document run as a method used for testing.

File: src/v2/providers/https.ts [hunk 5]
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
...
   run(data: CallableRequest<T>): Return;
 }
+/**
+ * Handle HTTPS requests
```
Comment (inlined) [inline src/v2/providers/https.ts:72]: Handles (Imagine the messages stars with "this method")

File: src/v2/providers/https.ts [hunk 6]
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
...
+ * @param opts Options to set on this Cloud Function
```
Comment (inlined) [inline src/v2/providers/https.ts:73]: params have a dash after the name

File: src/v2/providers/https.ts [hunk 7]
```diff
@@ -41,28 +41,49 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
...
+ * @param handler A function that takes a request and response object, same signature as an Express app.
+ */
```
Comment (inlined) [inline src/v2/providers/https.ts:75]: Please add a @returns.

File: src/v2/providers/https.ts [hunk 8]
```diff
@@ -41,28 +41,57 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
...
+   * If this is a String or ReqExp, allows requests from domains that match the provided string or RegExp.
+   * If this is an Array, allows requests from domains matching at least one entry of the array.
+   * Defaults to true for {@link CallableFunction}s.
+   */
   cors?: string | boolean | RegExp | Array<string | RegExp>;
 }
 
+/**
+ * Handles HTTPS requests.
+ */
```
Comment (TheIronDev) [inline src/v2/providers/https.ts:56]: Please document req/res

File: src/v2/providers/https.ts [hunk 9]
```diff
@@ -41,28 +41,57 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
  * Options that can be set on an individual HTTPS Cloud Function.
  */
 export interface HttpsOptions extends Omit<GlobalOptions, 'region'> {
-  /* HTTP functions can override and specify more than one regions. */
+  /** HTTP functions can override global options and can specify multiple regions to deploy to. */
   region?: SupportedRegion | string | Array<SupportedRegion | string>;
+  /** If true, allows CORS on requests to this function.
+   * If this is a String or ReqExp, allows requests from domains that match the provided string or RegExp.
```
Comment (egilmorez) [inline src/v2/providers/https.ts:47]: Should we backtick String and RegExp (as contrasted with the lowercase "the string" below?  

Suggest "If set to `String` or . . ."  and "If set to `Array`, ..."

File: src/v2/providers/https.ts [hunk 10]
```diff
@@ -41,28 +41,57 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
...
+   * If this is an Array, allows requests from domains matching at least one entry of the array.
+   * Defaults to true for {@link CallableFunction}s.
```
Comment (egilmorez) [inline src/v2/providers/https.ts:49]: Is it obvious from the code that it defaults to "false" for HTTP triggers, or should we add that?

File: src/v2/providers/https.ts [hunk 11]
```diff
@@ -41,28 +41,57 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
...
+   * @param data - An input for the handler function.
+   * @returns The output of the handler function.
+   */
   run(data: CallableRequest<T>): Return;
 }
+/**
+ * Handles HTTPS requests.
+ * @param opts - Options to set on this Cloud Function
+ * @param handler - A function that takes a {@link Request} and response object, same signature as an Express app.
+ * @returns A Cloud Function that you can export and deploy.
```
Comment (egilmorez) [inline src/v2/providers/https.ts:81]: Suggest "A function that you can..."

File: src/v2/providers/https.ts [hunk 12]
```diff
@@ -41,28 +41,57 @@ export { Request, CallableRequest, FunctionsErrorCode, HttpsError };
...
   opts: HttpsOptions,
   handler: (
     request: Request,
     response: express.Response
   ) => void | Promise<void>
 ): HttpsFunction;
+/**
+ * Handles HTTPS requests.
+ * @param handler - A function that takes a {@link Request} and response object, same signature as an Express app.
+ * @returns A Cloud Function that you can export and deploy.
```
Comment (egilmorez) [inline src/v2/providers/https.ts:93]: "A function that you can..."

File: src/v2/providers/https.ts [hunk 13]
```diff
@@ -160,10 +189,21 @@ export function onRequest(
   return handler as HttpsFunction;
 }
 
+/**
+ * Declares a callable method for clients to call using a Firebase SDK.
+ * @param opts - Options to set on this Cloud Function.
+ * @param handler - A function that takes a {@link CallableRequest}.
+ * @returns A Cloud Function that you can export and deploy.
```
Comment (egilmorez) [inline src/v2/providers/https.ts:196]: "A function that you can..."

File: src/v2/providers/https.ts [hunk 14]
```diff
@@ -160,10 +189,21 @@ export function onRequest(
   return handler as HttpsFunction;
 }
 
+/**
+ * Declares a callable method for clients to call using a Firebase SDK.
+ * @param opts - Options to set on this Cloud Function.
```
Comment (egilmorez) [inline src/v2/providers/https.ts:194]: Suggest "this function."

File: src/v2/providers/https.ts [hunk 15]
```diff
@@ -160,10 +189,21 @@ export function onRequest(
...
+ * @returns A Cloud Function that you can export and deploy.
+ */
 export function onCall<T = any, Return = any | Promise<any>>(
   opts: HttpsOptions,
   handler: (request: CallableRequest<T>) => Return
 ): CallableFunction<T, Return>;
+/**
+ * Declares a callable method for clients to call using a Firebase SDK.
+ * @param handler - A function that takes a {@link CallableRequest}.
+ * @returns A Cloud Function that you can export and deploy.
```
Comment (egilmorez) [inline src/v2/providers/https.ts:205]: "A function that you can..."

**Review by egilmorez — COMMENTED** [review]
Some thoughts Joe, thanks!

Kudos for staging and eyeballing the lists (I'm assuming you did that).

**Comment by joehan** [comment]
@egilmorez Addressed feedback. I haven't staged the list changes yet - I just manually reviewed the MD files and verified that the dcogen added linebreaks where expected.

**Review by egilmorez — APPROVED** [review]
LG, thanks!

**Pull request merged**